File: 백준/Silver/4659. 비밀번호 발음하기/비밀번호 발음하기.cc
#include <bits/stdc++.h>
using namespace std;

string word;

bool is_vowel(char c) {
	return (c == 'a') || (c == 'e') || (c == 'i') || (c == 'o') || (c == 'u');
}

bool is_acceptable(string str) {
	bool is_aeiou = false;
	bool isvowel = false;
	int cnt = 0;
	char prev;

	for (char c : str) {
		if (prev == c && (prev != 'e' && prev != 'o')) {
			return false;
		}

		if (is_vowel(c)) {
			is_aeiou = true;
			if (isvowel) {
				cnt++;

				if (cnt == 3)
					return false;
			}
			else {
				isvowel = true;
				cnt = 1;
			}
		}
		else {
			if (isvowel) {
				isvowel = false;
				cnt = 1;
			}
			else {
				cnt++;

				if (cnt == 3)
					return false;
			}
		}

		prev = c;
	}

	return is_aeiou;
}

int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(NULL); cout.tie(NULL);
    
	while (true) {
		cin >> word;

		if (word.compare("end") == 0)
			break;

		if (is_acceptable(word)) {
			cout << "<" << word << "> is acceptable.\n";
		}
		else {
			cout << "<" << word << "> is not acceptable.\n";
		}
	}

	return 0;
}

File: 백준/Silver/4659. 비밀번호 발음하기/비밀번호 발음하기.py
def is_acceptable(text):
    cnt = 0
    is_vowel = False
    has_aeiou = False
    prev = ''

    for c in text:
        if prev == c and c not in {'e', 'o'}:
            return False
        
        if c in {'a', 'e', 'i', 'o', 'u'}:
            has_aeiou = True

            if is_vowel:
                cnt += 1
                if cnt == 3:
                    return False
            else:
                is_vowel = True
                cnt = 1

        else:
            if is_vowel:
                is_vowel = False
                cnt = 1
            else:
                cnt += 1
                if cnt == 3:
                    return False
                
        prev = c

    return has_aeiou

while True:
    text = input().strip()

    if text == "end":
        break

    if is_acceptable(text):
        print(f"<{text}> is acceptable.")
    else:
        print(f"<{text}> is not acceptable.")
